### PR TITLE
Add separate doc for GitHub integration

### LINKED ISSUE #313: Add documentation for non-enterprise Github integration set up
Currently the only docs we have for Github integration set up focuses on Github Enterprise setup. Can we add an article that goes over the steps to integrate with a regular Github org?

### PR DESCRIPTION
**Pull request by heitortsergent**
Fixes #313.

I'm thinking we could also expand this change so each provider page covers the regular integration + self-hosted instructions. I think it can be potentially confusing if a user just jumps in straight to the `GitLab` page and sees the note at the top that says it's only available in the Pro plan and above.

@pamgoodrich what do you think?

### CODE REVIEW

**Comment by pamgoodrich** [comment]
> Fixes #313.
> 
> I'm thinking we could also expand this change so each provider page covers the regular integration + self-hosted instructions. I think it can be potentially confusing if a user just jumps in straight to the `GitLab` page and sees the note at the top that says it's only available in the Pro plan and above.
> 
> @pamgoodrich what do you think?

I think that works. It might be good to run it by @chohmann since she's pretty familiar with this content and how customers use it.

File: 2.-workspaces/configure-git/g.github.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+1. In the upper-right corner of any page, select your profile photo, then select **Settings**.
+2. In the left sidebar, select **Developer settings**.
+3. Select  **OAuth Apps**.
+4. Select **New OAuth App**.
+    - If you haven't created an app before, this button is labeled as **Register a new application**.
+5. Add the following information:
+   * **Application name**: Add the name of your app (e.g `Stoplight Integration`).
+   * **Homepage URL**: Add the full URL to your app's website (e.g.: `https://stoplight.io`).
+   * **Authorization callback URL**: Set as `https://stoplight.io/oauth/callback`.
```
Comment (chohmann) [inline 2.-workspaces/configure-git/g.github.md:34]: I wonder if it's better to mention grabbing this from the integration settings form?

image

That way if it ever changes or something, these docs don't need to change. Also on prem customers would not set this to `https://stoplight.io/oauth/callback`, they'd would need to set it wherever they're hosting their stoplight instance. We calculate that for them and display that in the Redirect URI slot.

File: 2.-workspaces/configure-git/g.github.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+stoplight-id: dhre9np9qe1dc
+---
+
+# GitHub
+
+GitHub is supported out of the box in Stoplight. To enable your workspace GitHub integration:
```
Comment (chohmann) [inline 2.-workspaces/configure-git/g.github.md:7]: I like the way this is setup for explaining being able to use the default configuration for any plan vs. the customized setup for Pro and above. 

I think it'd be a good addition to have this for each integration that supports default settings:
- GitLab
- Bitbucket Cloud
- Azure DevOps Services (potentially. This is going to be added in the next couple sprints I think from burrito dogs)

File: 2.-workspaces/configure-git/g.github.md (hunk 1 above)
Comment (heitortsergent) [reply 2.-workspaces/configure-git/g.github.md]: That's a great point @chohmann. I made some small changes to those steps, do you think it works better now?

File: 2.-workspaces/configure-git/g.github.md (hunk 2 above)
Comment (heitortsergent) [reply 2.-workspaces/configure-git/g.github.md:7]: I made a separate PR #354 to replicate the changes to the other providers :D. I figured I could get this added as a small change before touching on all the other pages.

Comment (chohmann) [reply 2.-workspaces/configure-git/g.github.md:7]: niiiiice 🙌

File: 2.-workspaces/configure-git/g.github.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+1. Select **Settings** from the workspace home page, then navigate to the **Integrations** section.
+2. Select **Add** for GitHub.
+3. Select **Customize integration settings**.
+4. Add the following information:
+   - **Host URL**: Set as `https://github.com`.
+   - **API URL**: Set as `https://api.github.com`.
+   - **Client ID**: You can find this as `Client ID` under the OAuth Apps settings for the newly created app.
+   - **Client Secret**: You can find this as `Client Secret` under the OAuth consumers settings for the newly created app.
+   - **Redirect URL**: Copy this URL, and in your newly created GitHub OAuth app, paste it in the `Authorization callback URL` field and select **Update application**.
```
Comment (heitortsergent) [inline 2.-workspaces/configure-git/g.github.md:50]: @chohmann I realized my last comment didn't show up on the updated section. 😅 

Does this work better for the Redirect URL instructions? I also updated line 34 here.

Comment (chohmann) [reply 2.-workspaces/configure-git/g.github.md:50]: Yes I like this! 👍

**Review by chohmann — APPROVED** [review]
🚀

**Pull request merged**